File: grass-next-prj/GrassNext_project/GrassNextFrontend/src/app/model/contours.ts
/**
 * Represents the result of a contour processing operation.
 **
 * @property {Contour[]} contours - An array of Contour objects representing the detected contours.
 * @property {number} duration - The time taken to process the contours, typically in milliseconds.
 * @property {boolean} error - Indicates whether an error occurred during processing.
 * @property {string} msg - A message providing details about the operation or any encountered error.
 */
export interface Contours {
  contours: Contour[];
  duration: number;
  error: boolean;
  msg: string;
}

/**
 * Represents a contour line or shape defined by points and associated with a specific color and threshold.
 *
 * @property {string} color - The visual representation color of the contour.
 * @property {Point[]} points - An array of points defining the vertices or defining positions of the contour.
 * @property {string} threshold - A value representing the threshold level associated with the contour.
 *
 */
export interface Contour {
  color: string;
  points: Point[];
  threshold: string;
}

/**
 * Represents a geographical point with latitude and longitude coordinates.
 *
 * @property {number} lat - The latitude of the point.
 * @property {number} lon - The longitude of the point.
 *
 */
export interface Point {
  lat: number;
  lon: number;
}

File: grass-next-prj/GrassNext_project/GrassNextServer/src/main/java/com/grassnext/grassnextserver/common/Contours.java
package com.grassnext.grassnextserver.common;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * Represents a collection of contours along with additional metadata about the processing status.
 *
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Contours {
    /**
     * Inner class representing a singular point on the map.
     *
     */
    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class GpsPoint {
        /**
         * Latitude of the point on the map.
         *
         */
        public double lat;
        /**
         * Longitude of the point on the map.
         *
         */
        public double lon;
    };

    /**
     * Inner class representing a contour on the map.
     *
     */
    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class Contour {
        /**
         * Threshold for which the contour is created.
         *
         */
        public String threshold;
        /**
         * Color of the contour.
         *
         */
        public String color;
        /**
         * List of {@link GpsPoint} elements that the contour consist of.
         *
         */
        public List<GpsPoint> points;
    };

    /**
     * A boolean flag indicating whether an error occurred during the processing of the contours.
     *
     */
    public boolean error;
    /**
     * A string variable used to store descriptive messages.
     *
     */
    public String msg;
    /**
     * Represents the duration of the contour processing operation.
     *
     */
    public double duration;
    /**
     * A list of contours created for the map
     *
     */
    public List<Contour> contours;
}
